vcf-dumper/vcf-dumper-ws/src/main/java/uk/ac/ebi/eva/vcfdump/server/HtsgetVcfController.java

```diff
@@ -0,0 +1,197 @@
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MultivaluedHashMap;
+import javax.ws.rs.core.MultivaluedMap;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Properties;
+
+@RestController
+@RequestMapping(value = "/variants/")
+@Api(tags = {"htsget"})
+public class HtsgetVcfController {
+
+    private static final String VCF = "VCF";
+
+    private Properties evaProperties;
+
+    public HtsgetVcfController() throws IOException {
+        evaProperties = new Properties();
+        evaProperties.load(VcfDumperWSServer.class.getResourceAsStream("/eva.properties"));
+    }
+
+    @RequestMapping(value = "/{id}", method = RequestMethod.GET, consumes = "application/*",
+            produces = "application/vnd.ga4gh.htsget.v0.2rc+json; charset=UTF-8")
+    public ResponseEntity getHtsgetUrls(
+            @PathVariable("id") String id,
+            @RequestParam(name = "format", required = false) String format,
+            @RequestParam(name = "referenceName", required = false) String referenceName,
+            @RequestParam(name = "species", required = false) String species,
+            @RequestParam(name = "start", required = false) Integer start,
+            @RequestParam(name = "end", required = false) Integer end,
+            @RequestParam(name = "fields", required = false) List<String> fields,
+            @RequestParam(name = "tags", required = false, defaultValue = "") String tags,
+            @RequestParam(name = "notags", required = false, defaultValue = "") String notags,
+            HttpServletRequest request,
+            HttpServletResponse response)
+            throws IllegalAccessException, IllegalOpenCGACredentialsException, InstantiationException, IOException,
+            StorageManagerException, URISyntaxException, ClassNotFoundException {
+
+        if (!VCF.equals(format)) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(Collections.singletonMap("htsget",
+                            new HtsGetError("UnsupportedFormat", "Specified format is not supported by this server")));
+        }
+
+        String dbName = "eva_" + species;
+        MultivaluedMap<String, String> queryParameters = new MultivaluedHashMap<>();
+        queryParameters.put(VariantDBAdaptor.REGION, Collections.singletonList(referenceName));
+
+        int blockSize = Integer.parseInt(evaProperties.getProperty("eva.htsget.blocksize"));
+
+        VariantExporterController controller = new VariantExporterController(dbName, Arrays.asList(id.split(",")),
+                                                                             evaProperties,
+                                                                             queryParameters, blockSize);
+        if (!controller.validateSpecies()) {
```

Comment:
I would move those three validations (species, studies and referenceName) to a separated method:


```
    ResponseEntity errorResponse =  validateRequest(controller, referenceName);
    if (errorResponse != null) {
          return errorResponse;
    } else {
    ... continue method Execution here
}
private ResponseEntity validateRequest(VariantExporterController controller, String referenceName) {...}```